Show bulk actions below table

howdu opened the pull request:
## Description
Add option to display bulk actions below table

```php
->bulkActions([
])
->bulkActionsPosition(Tables\Enums\BulkActionsPosition::BelowTable)
```

## Visual changes
<img width="852" alt="Screenshot 2024-07-10 at 13 38 58" src="https://github.com/filamentphp/filament/assets/533658/ba770d85-db58-4478-92a7-b7d3e5db7223">

<img width="1226" alt="Screenshot 2024-07-10 at 13 35 58" src="https://github.com/filamentphp/filament/assets/533658/a4c3d71c-a57a-476f-8ac4-1cbcf69a38a6">

<!-- Add screenshots/recordings of before and after. -->

## Functional changes

- [x] Code style has been fixed by running the `composer cs` command.
- [x] Changes have been tested to not break existing functionality.
- [x] Documentation is up-to-date.

howdu committed:
Add bulkActionsPosition to display below table

howdu committed:
chore: fix code style

howdu committed:
Update 05-actions.md

danharrin commented:
@zepfietje would appreciate the green light on the design before I proceed into the code please

zepfietje commented:
Personally I don't think having the actions at the bottom results in a good UI/UX.
Could you clarify your reasoning here, @howdu?

howdu commented:
@zepfietje it was a client who requested it - their reasoning was they work their way down the table selecting all the relevant records and always finish at the bottom where they wanted the action button instead of having to scroll to the top.

The table is implemented as a checklist where they need to work through all the rows selecting all the relevant records at once.

zepfietje commented:
Hmm, I'm wondering if the bulk select checkbox and select/deselect actions should be in the footer too then?

howdu committed:
Merge branch '3.x' into feature/bulk-actions-below-table

howdu committed:
Add bulk above and below option. Add footer bulk selections

howdu committed:
chore: fix code style

howdu commented:
@zepfietje I've added this above the action let me know what you think.
<img width="386" alt="Screenshot 2024-07-21 at 07 45 37" src="https://github.com/user-attachments/assets/b945488f-266c-42fd-9ce3-9cb0fc1f45ab">

<img width="761" alt="Screenshot 2024-07-21 at 07 45 54" src="https://github.com/user-attachments/assets/eac5973e-aad9-402e-91d9-6515f3d5d58d">

Also added the option to display both above and below table. You can get clever with not displaying this on a small tables.
```php
->bulkActionsPosition(
    fn (Table $table) => $table->getRecords()->count() > 5
        ? Tables\Enums\BulkActionsPosition::AboveAndBelowTable
        : Tables\Enums\BulkActionsPosition::AboveTable
)
```

zepfietje commented on packages/tables/resources/views/index.blade.php:
```diff
@@ -1250,6 +1254,36 @@ class="fi-ta-record-checkbox"
         </div>
 
+        @if ($hasBulkActionsBelowTable && (! $isReordering) && count($bulkActions))
+            <div
```
Is there a way to minimize duplication here and in the header?
Maintenance of the table index view is already very hard, so I don't want to introduce additional complexity.

howdu replied:
@zepfietje please let me know what changes you think best to improve this. 

Perhaps using `capture` blade directive? But due to it not including the reorder action in the footer I think the `selection-indicator` and bulk action`filament-tables::actions` would need to be captured separately?

zepfietje commented:
@howdu, shouldn't it also show the bulk selection checkbox?
It should be identical to what's in the header, right?

howdu commented:
@zepfietje I think adding the bulk checkbox would overcomplicating this. I was trying to keep it simple and work the same as filter position above/below.

zepfietje commented:
Alright, I understand now since the bulk selection checkbox is in the column headers row.
What does this PR look like when positioning bulk actions *and* filters in the footer of the table, @howdu?

danharrin commented:
Going to close this as it feels stale. I'm not entirely sure I like how the bulk actions look under the table and I can't say I've had many requests for it either. If there is a particular explanation behind it, let me know and we can consider accepting a PR in v4.
